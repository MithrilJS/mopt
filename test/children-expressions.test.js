"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("Expression children", () => {
        it("should support expressions with literal values (strings)", () =>
            expect(
                code(`m("0", "1" + "2")`)
            )
            .toMatchSnapshot()
        );

        it("should support expressions with literal values (numbers)", () =>
            expect(
                code(`m("0", 1 + 2)`)
            )
            .toMatchSnapshot()
        );

        it("should support expressions with literal values (strings + numbers)", () =>
            expect(
                code(`m("0", 1 + "2")`)
            )
            .toMatchSnapshot()
        );

        it("should convert expressions containing more than 2 values", () =>
            expect(
                code(`m("0", "1" + "2" + "3")`)
            )
            .toMatchSnapshot()
        );

        it("should convert expressions with operators other than +", () =>
            expect(
                code(`m("0", 1 - 2)`)
            )
            .toMatchSnapshot()
        );
        
        it("should not convert non-literal values", () =>
            expect(
                code(`m("0", a + b)`)
            )
            .toMatchSnapshot()
        );
    });
});
