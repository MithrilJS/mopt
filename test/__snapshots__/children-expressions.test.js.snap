// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children Expression children should convert expressions containing more than 2 values 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,\\"1\\"+\\"2\\"+\\"3\\",undefined);"`;

exports[`Children Expression children should convert expressions with operators other than + 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,1-2,undefined);"`;

exports[`Children Expression children should not convert non-literal values 1`] = `"m(\\"0\\",a+b);"`;

exports[`Children Expression children should support expressions with literal values (numbers) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,1+2,undefined);"`;

exports[`Children Expression children should support expressions with literal values (strings + numbers) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,1+\\"2\\",undefined);"`;

exports[`Children Expression children should support expressions with literal values (strings) 1`] = `"m.vnode(\\"0\\",undefined,undefined,undefined,\\"1\\"+\\"2\\",undefined);"`;
